R/auth.R

```diff
@@ -0,0 +1,24 @@
+#' @title Get the API key
+#' @description Get the API key from the environment variable DELPHI_EPIDATA_KEY or getOption("delphi.epidata.key")
+#' @param verbose Will print the source of the key if TRUE (e.g. options, environment, or config file)
+#' @return The API key
+#' @export
+get_auth_key <- function() {
+  key <- Sys.getenv("DELPHI_EPIDATA_KEY", unset = "")
+  if (key != "") return(key)
+
+  key <- getOption("delphi.epidata.key", default = "")
+  if (key != "") return(key)
+
+  warning(
+    paste0(
+      "No API key found. To avoid being rate limited, you can:\n",
+      " - set the environment variable DELPHI_EPIDATA_KEY.\n",
+      " - set the option 'delphi.epidata.key' or\n",
+      "\n",
+      "To save your key for later sessions (and hide it from your code), you can edit your .Renviron file with:\n",
+      "  usewith::edit_r_environ()"
+    )
+  )
```

Comment:
suggestion: we could buy into `cli` here to pretty bulleted lists with auto line wrap.  Think it was `cli::cli_bullets`?  I'm not up to speed on all the `cli` features we might consider using; @dajmcdon might have a better idea.  Could be a good topic to go over for a few minutes at a tooling meeting: where / for what we'd want to use `cli` / the most useful functions.

Reply:
also of interest: using https://cli.r-lib.org/reference/links.html to link directly to page to get a key?